clang/lib/CIR/CodeGen/CIRGenExprScalar.cpp

```diff
@@ -1484,6 +1484,20 @@ mlir::Value ScalarExprEmitter::VisitUnaryLNot(const UnaryOperator *e) {
   return {};
 }
 
+/// If the specified expression does not fold
+/// to a constant, or if it does but contains a label, return false.  If it
+/// constant folds return true and set the boolean result in Result.
+bool CIRGenFunction::constantFoldsToSimpleInteger(const Expr *cond,
```

Comment:
Can you rename this to `constantFoldsToBool` and move it to the same file as the other `constantFoldsToInteger` implementation?